src/FixMath.h

```diff
@@ -468,55 +455,48 @@ class UFix
       @return true if this equal to op, false otherwise
   */
   template<int8_t _NI, int8_t _NF>
-  bool operator== (const UFix<_NI,_NF>& op) const
+  constexpr bool operator== (const UFix<_NI,_NF>& op) const
   {
-    constexpr int8_t new_NI = FixMathPrivate::FM_max(NI, _NI);
-    constexpr int8_t new_NF = FixMathPrivate::FM_max(NF, _NF);
-    UFix<new_NI,new_NF> left(*this);
-    UFix<new_NI,new_NF> right(op);
-    return left.asRaw()==right.asRaw();
+    using namespace FixMathPrivate;
+    typedef UFix<FM_max(NI, _NI),FM_max(NF, _NF)> comptype;  // type suitable for comparison
+    return (comptype(*this).asRaw()==comptype(op).asRaw());
   }
 
   /** Comparison with another UFix.
       @param op A UFix
       @return true if this not equal to op, false otherwise
   */
   template<int8_t _NI, int8_t _NF>
-  bool operator!= (const UFix<_NI,_NF>& op) const
+  constexpr bool operator!= (const UFix<_NI,_NF>& op) const
   {
-    constexpr int8_t new_NI = FixMathPrivate::FM_max(NI, _NI);
-    constexpr int8_t new_NF = FixMathPrivate::FM_max(NF, _NF);
-    UFix<new_NI,new_NF> left(*this);
-    UFix<new_NI,new_NF> right(op);
-    return left.asRaw()!=right.asRaw();
+    return !(*this==op);
```

Comment:
Ah yes, indeed… But I always wonder if `!=` and `==` are equally fast: for the former, once one bit is found different there is no need to go further to know they are different, whereas for the latter the processor needs to check them all. Note that it is likely that it is done in one shot anyway so if an optimization it is, it is probably a micro one…

Reply:
A valid point. I would hope for this to be optimized, but I am not sure. Since it's not a lot of trouble, I reverted these to the original logic (in new formulation).